src/main/java/dev/enjarai/trickster/spell/trick/block/WeatherTrick.java

```diff
@@ -0,0 +1,91 @@
+package dev.enjarai.trickster.spell.trick.block;
+
+import dev.enjarai.trickster.spell.Fragment;
+import dev.enjarai.trickster.spell.Pattern;
+import dev.enjarai.trickster.spell.SpellContext;
+import dev.enjarai.trickster.spell.blunder.BlockInvalidBlunder;
+import dev.enjarai.trickster.spell.blunder.BlunderException;
+import dev.enjarai.trickster.spell.fragment.FragmentType;
+import dev.enjarai.trickster.spell.trick.Trick;
+import net.minecraft.block.*;
+import net.minecraft.fluid.Fluids;
+import net.minecraft.registry.Registries;
+import net.minecraft.registry.RegistryKeys;
+import net.minecraft.registry.entry.RegistryEntry;
+import net.minecraft.registry.tag.TagKey;
+import net.minecraft.server.world.ServerWorld;
+import net.minecraft.state.property.Properties;
+import net.minecraft.util.math.random.Random;
+import net.minecraft.world.WorldEvents;
+import net.minecraft.world.event.GameEvent;
+
+import java.util.List;
+import java.util.Optional;
+
+public class WeatherTrick extends Trick {
+    public WeatherTrick() {
+        super(Pattern.of(0, 4, 6, 7, 8, 4, 2));
+    }
+
+    @Override
+    public Fragment activate(SpellContext ctx, List<Fragment> fragments) throws BlunderException {
+        var weatheringPosFragment = expectInput(fragments, FragmentType.VECTOR, 0);
+        var weatheringPos = weatheringPosFragment.toBlockPos();
+        var waterPos = expectInput(fragments, FragmentType.VECTOR, 1).toBlockPos();
+
+        expectCanBuild(ctx, weatheringPos);
+        expectCanBuild(ctx, waterPos);
+
+        ServerWorld world = ctx.source().getWorld();
+
+        BlockState blockState = world.getBlockState(weatheringPos);
+
+        BlockState state = world.getBlockState(waterPos);
+
+        if ((!state.isOf(Blocks.WATER_CAULDRON) || state.get(LeveledCauldronBlock.LEVEL) != 3) && !state.getFluidState().isOf(Fluids.WATER)) {
+            throw new BlockInvalidBlunder(this);
+        }
```

Comment:
Why this?

Reply:
Makes sure you are providing a water source to use for the spell
it originally used only cauldrons to force the use of spells to provide the water but that was reduced for balancing

Reply:
I mean this isn't really *important*, water is pretty much free in Minecraft, but it'd make using it in the Nether more annoying, which is interesting, if anything.

Reply:
It's just a remainder of the original concept, I can remove it, but it does allow me to add it to the description that I throw a lot of water onto the block